File: image-smoother.js
var imageSmoother = function(img) {
  const m = img.length;
  const n = img[0].length;
  const result = Array.from({ length: m }, () => Array(n).fill(0));
  const directions = [
      [0, 0], [0, 1], [0, -1],
      [1, 0], [1, 1], [1, -1],
      [-1, 0], [-1, 1], [-1, -1]
  ];

  for (let i = 0; i < m; i++) {
      for (let j = 0; j < n; j++) {
          let neighborSum = 0;
          let neighborCount = 0;

          for (const [dx, dy] of directions) {
              const x = i + dx;
              const y = j + dy;

              if (0 <= x && x < m && 0 <= y && y < n) {
                  neighborSum += img[x][y];
                  neighborCount++;
              }
          }

          result[i][j] = Math.floor(neighborSum / Math.max(1, neighborCount));
      }
  }

  return result;
}

File: image-smoother.py
class Solution:
    def imageSmoother(self, img: List[List[int]]) -> List[List[int]]:
        m, n = len(img), len(img[0])
        result = [[0] * n for _ in range(m)]

        directions = [(0, 0), (0, 1), (0, -1), (1, 0), (1, 1), (1, -1), (-1, 0), (-1, 1), (-1, -1)]

        for i in range(m):
            for j in range(n):
                neighbors_sum, neighbors_count = 0, 0

                for dx, dy in directions:
                    x, y = i + dx, j + dy
                    if 0 <= x < m and 0 <= y < n:
                        neighbors_sum += img[x][y]
                        neighbors_count += 1

                result[i][j] = neighbors_sum // neighbors_count

        return result